In [9]:
import time

In [5]:
start = time.time()
!python train.py --epochs 5
end = time.time() - start

train: weights=yolov5s.pt, cfg=, data=data\coco128.yaml, hyp=data\hyps\hyp.scratch-low.yaml, epochs=5, batch_size=16, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, noplots=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs\train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[0], save_period=-1, seed=0, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
github: up to date with https://github.com/ultralytics/yolov5 
YOLOv5  v7.0-128-gb96f35c Python-3.9.2 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3060, 12287MiB)

hyperparameters: lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=

                 Class     Images  Instances          P          R      mAP50   mAP50-95:  25%|##5       | 1/4 [00:00<00:01,  2.62it/s]
                 Class     Images  Instances          P          R      mAP50   mAP50-95:  50%|#####     | 2/4 [00:00<00:00,  2.81it/s]
                 Class     Images  Instances          P          R      mAP50   mAP50-95:  75%|#######5  | 3/4 [00:01<00:00,  2.69it/s]
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|##########| 4/4 [00:01<00:00,  2.70it/s]
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|##########| 4/4 [00:01<00:00,  2.70it/s]
                   all        128        929      0.722      0.627      0.707       0.47

      Epoch    GPU_mem   box_loss   obj_loss   cls_loss  Instances       Size

  0%|          | 0/8 [00:00<?, ?it/s]
        2/4      4.51G    0.04367     0.0725    0.01565        219        640:   0%|          | 0/8 [00:00<?, ?it

In [6]:
tep = divmod(end, 60)

In [7]:
print(tep)

(2.0, 13.689990520477295)


In [10]:
start1 = time.time()
!python detect.py --weights yolov5s.pt --source data/test 
end1 = time.time() - start1

detect: weights=['yolov5s.pt'], source=data/test, data=data\coco128.yaml, imgsz=[640, 640], conf_thres=0.25, iou_thres=0.45, max_det=1000, device=, view_img=False, save_txt=False, save_conf=False, save_crop=False, nosave=False, classes=None, agnostic_nms=False, augment=False, visualize=False, update=False, project=runs\detect, name=exp, exist_ok=False, line_thickness=3, hide_labels=False, hide_conf=False, half=False, dnn=False, vid_stride=1
YOLOv5  v7.0-128-gb96f35c Python-3.9.2 torch-2.0.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3060, 12287MiB)

Fusing layers... 
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients
image 1/10 C:\Users\rhayr\Downloads\Compressed\yolov5\data\test\000000000139.jpg: 448x640 1 person, 1 bottle, 3 chairs, 2 potted plants, 3 dining tables, 2 tvs, 3 refrigerators, 1 clock, 4 vases, 72.0ms
image 2/10 C:\Users\rhayr\Downloads\Compressed\yolov5\data\test\000000000285.jpg: 640x608 1 bear, 65.0ms
image 3/10 C:\Users\rhayr\Downloads\Compressed\yolov5\data\test\000

Speed: 0.6ms pre-process, 46.7ms inference, 1.8ms NMS per image at shape (1, 3, 640, 640)
Speed: 0.4ms pre-process, 42.2ms inference, 2.1ms NMS per image at shape (1, 3, 640, 640)
Speed: 0.8ms pre-process, 43.1ms inference, 2.0ms NMS per image at shape (1, 3, 640, 640)
